internal/report/json/json.go

```diff
@@ -54,6 +55,20 @@ func (r *Reporter) Render() {
 	}
 }
 
+// Remove empty values from the data to clean it up.
+func (r *Reporter) cleanup() {
+	for _, r := range r.Results {
+		var artifacts []report.Artifact
+		for _, a := range r.Artifacts {
+			if a.FilePath == "" {
+				continue
+			}
+			artifacts = append(artifacts, a)
+		}
+		r.Artifacts = artifacts
```

Comment:
Could you elaborate more as to why this is needed? 

1. How is it that we have artifacts that don't have a filepath? Is that a mistake made somewhere else?
2. Why is it cleaned up here?

Reply:
I found this by enabling both `junit` and `json` reporter. When adding junit artifact here https://github.com/saucelabs/saucectl/pull/812/files#diff-0a4ce723eaf8f4121ec686bc22e0f79d414c6b5bde396ef60b99eab8435bd550R493
It doesn't have `filePath` thus causing an empty items.

Another way is to retrieve `fileName` of the reporter and assign it here. Is it a better solution?

Reply:
And another question. Is a JUnit report file or JSON report file considered an artifact in the JSON report?

Reply:
I'm not entirely sure what the real question is 😅 

All artifacts are artifacts in the eyes of any reporter. The reporter merely reports what artifacts have been downloaded, [see description](https://github.com/saucelabs/saucectl/blob/8df4b5ddaba2d01d93b9cf287bd199b7752b76ce/internal/report/report.go#L35-L35).

Some reporters require specific artifacts to work properly, like the junit reporter.

Having said that, I don't know why the json reporter says [it needs json artifacts](https://github.com/saucelabs/saucectl/blob/8df4b5ddaba2d01d93b9cf287bd199b7752b76ce/internal/report/json/json.go#L64-L64). Looks like another code smell or bug.

Reply:
In our implementation, we've added the [JUnit file](https://github.com/saucelabs/saucectl/pull/812/files#diff-0a4ce723eaf8f4121ec686bc22e0f79d414c6b5bde396ef60b99eab8435bd550R493) as an artifact, but it doesn't have any real downloaded file path. This is the reason why I do the cleanup.

When considering if I can use the [JUnit report filename](https://github.com/saucelabs/saucectl/blob/8df4b5ddaba2d01d93b9cf287bd199b7752b76ce/internal/config/config.go#L148) as the file path, the question arose about whether [JSON report file](https://github.com/saucelabs/saucectl/blob/8df4b5ddaba2d01d93b9cf287bd199b7752b76ce/internal/config/config.go#L154) and [JUnit report file](https://github.com/saucelabs/saucectl/blob/8df4b5ddaba2d01d93b9cf287bd199b7752b76ce/internal/config/config.go#L148) should be considered artifacts.

If yes, then they should be included in the JSON report file as follows:
```
"artifacts": [
      {
        "filePath": "my_junit_report.xml"
      },
      {
        "filePath": "my_json_report.json"
      },
      {
        "filePath": "artifacts/Firefox_Win.3/junit.xml"
      },
...
]
```

If no, then they should not be included in the JSON report:
```
"artifacts": [
      {
        "filePath": "artifacts/Firefox_Win.3/junit.xml"
      },
...
```
Considering that the [JUnit file](https://github.com/saucelabs/saucectl/pull/812/files#diff-0a4ce723eaf8f4121ec686bc22e0f79d414c6b5bde396ef60b99eab8435bd550R493) is not the real downloaded one and based on the description and above discussion, I don't think they should be artifacts and included in the JSON report. Thus, the cleanup is necessary. The generated artifacts would look like:

```
"artifacts": [
      {
        "filePath": "artifacts/Firefox_Win.3/junit.xml"
      },
...
```

Reply:
Sure. Let me make it more clear.